### PR TITLE
rtx 1.35.6

### PR DESCRIPTION
**Pull request by jdx**
Created with `brew bump-formula-pr`.<details>
  <summary>release notes</summary>
  <pre>## What's Changed
* simplify build features by @jdxcode in https://github.com/jdxcode/rtx/pull/747

**Full Changelog**: https://github.com/jdxcode/rtx/compare/v1.35.5...v1.35.6</pre>
</details>

### CODE REVIEW

File: Formula/rtx.rb [hunk 1]
```diff
@@ -18,6 +18,10 @@ class Rtx < Formula
 
   depends_on "rust" => :build
 
+  on_linux do
+    depends_on "openssl@3" # Uses Secure Transport on macOS
+  end
```
Comment (chenrui333) [inline Formula/rtx.rb:23]: lack of `pkg-config` for linux build

File: Formula/rtx.rb [hunk 2]
```diff
@@ -18,6 +18,11 @@ class Rtx < Formula
 
   depends_on "rust" => :build
 
+  on_linux do
+    depends_on "pkg-config" => :build
```
Comment (jdxcode) [inline Formula/rtx.rb:22]: you sure this is required? sccache doesn't have it

Comment (jdxcode) [reply Formula/rtx.rb:22]: it does however set this which I was checking to see if was required or not:

```
    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix if OS.linux?
```

Comment (jdxcode) [reply Formula/rtx.rb:22]: looks like the build passed without it

Comment (chenrui333) [reply Formula/rtx.rb:22]: > you sure this is required? sccache doesn't have it

it does need it, it was actually alerting in this run, https://github.com/Homebrew/homebrew-core/actions/runs/5706888197/job/15464877308

```
    run pkg_config fail: Could not run `PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" "pkg-config" "--libs" "--cflags" "openssl"`
    The pkg-config command could not be found.
 ```

Comment (chenrui333) [reply Formula/rtx.rb:22]: > it does however set this which I was checking to see if was required or not:
> 
> ```
>     ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix if OS.linux?
> ```

I think it is fine.

Comment (chenrui333) [reply Formula/rtx.rb:22]: in the build log, I can see the linkage is done correctly
image

**Review by chenrui333 — APPROVED** [review]

**Review by jdx — COMMENTED** [review]
lgtm

**Pull request merged**